query/src/test/java/com/connexta/replication/query/QueryServiceTest.java

```diff
@@ -0,0 +1,574 @@
+    Metadata metadata = mockBasicMetadata();
+    QueryResponse queryResponse = mock(QueryResponse.class);
+    when(adapter.query(any(QueryRequest.class))).thenReturn(queryResponse);
+    when(queryResponse.getMetadata()).thenReturn(List.of(metadata));
+
+    // #metadataToTask
+    when(filter.getPriority()).thenReturn((byte) 0);
+    when(metadata.getResourceSize()).thenReturn(0L);
+    Instant lastModified = Instant.now();
+    when(metadata.getMetadataModified()).thenReturn(Date.from(lastModified));
+    ArgumentCaptor<Instant> lastModifiedCaptor = ArgumentCaptor.forClass(Instant.class);
+    when(metadata.getType()).thenReturn(HashMap.class);
+    when(metadata.getRawMetadata()).thenReturn(new HashMap<>());
+
+    queryService.query();
+
+    verify(adapter).query(requestArgumentCaptor.capture());
+    assertThat(requestArgumentCaptor.getValue().getModifiedAfter(), is(Date.from(index)));
+    verify(siteQueue).put(any(TaskInfoImpl.class));
+    verify(filterIndex).setModifiedSince(lastModifiedCaptor.capture());
+    assertThat(lastModifiedCaptor.getValue().isAfter(index), is(true));
+    verify(queryServiceTools).saveFilterIndex(filterIndex);
+  }
+
+  @Test
+  public void queryThrowsAdapterException() throws Exception {
+    // #query
+    when(adapter.isAvailable()).thenReturn(true);
+    Filter filter = mock(Filter.class);
+    when(queryServiceTools.activeFiltersFor(site)).thenReturn(Stream.of(filter));
+    FilterIndex filterIndex = mock(FilterIndex.class);
+    when(queryServiceTools.getOrCreateFilterIndex(filter)).thenReturn(filterIndex);
+
+    // #getNewMetadataForFilter
+    when(filterIndex.getModifiedSince()).thenReturn(Optional.empty());
+    when(adapter.query(any(QueryRequest.class))).thenThrow(new AdapterException(""));
+
+    queryService.query();
+
+    verify(siteQueue, never()).put(any(TaskInfoImpl.class));
+    verify(filterIndex).setModifiedSince(Instant.MIN);
+    verify(queryServiceTools).saveFilterIndex(filterIndex);
+  }
+
+  @Test
+  public void queryThrowsAdapterInterruptedException() throws Exception {
+    // #query
+    when(adapter.isAvailable()).thenReturn(true);
+    Filter filter = mock(Filter.class);
+    when(queryServiceTools.activeFiltersFor(site)).thenReturn(Stream.of(filter));
+    FilterIndex filterIndex = mock(FilterIndex.class);
+    when(queryServiceTools.getOrCreateFilterIndex(filter)).thenReturn(filterIndex);
+
+    // #getNewMetadataForFilter
+    when(filterIndex.getModifiedSince()).thenReturn(Optional.empty());
+    when(adapter.query(any(QueryRequest.class)))
+        .thenThrow(new AdapterInterruptedException(new InterruptedException()));
+
+    queryService.query();
+
+    verify(siteQueue, never()).put(any(TaskInfoImpl.class));
+    verify(filterIndex).setModifiedSince(Instant.MIN);
+    verify(queryServiceTools).saveFilterIndex(filterIndex);
+  }
+
+  @Test
+  public void queryThrowsInterruptedException() throws Exception {
```

Comment:
✏️ Maybe be clearer about the fact that you are testing interruptions when putting tasks in the queue